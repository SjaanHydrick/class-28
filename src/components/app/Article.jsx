import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, url, urlToImage, author, description }) => (
  <figure>
    <a href={url}><h2>{title}</h2></a>
    <img src={urlToImage} />
    <h3>{author}</h3>
    <p>{description}</p>
  </figure>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  url: PropTypes.string,
  urlToImage: PropTypes.string,
  description: PropTypes.string.isRequired
};

export default Article;
