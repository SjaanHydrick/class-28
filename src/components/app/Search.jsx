import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ handleSearch, handleClick }) => (
  <>
    <form onSubmit={handleClick}>
      <input
        type="text"
        onChange={handleSearch}
        data-testid="searchBar"
      />
      <button data-testid="submitButton">Search</button>
    </form>
  </>
);

Search.propTypes = {
  handleSearch: PropTypes.func,
  handleClick: PropTypes.func
};

export default Search;
