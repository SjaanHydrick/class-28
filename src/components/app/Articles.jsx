import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';
import styles from '../container/NewsSearch.css';

const Articles = ({ articles }) => {
  const articleElements = articles.map(article => (
    <div className={styles.article} key={article.title}>
      <Article {...article} />
    </div>
  ));

  return (
    <div>
      {articleElements}
    </div>
  );
};

Articles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string,
    author: PropTypes.string,
    url: PropTypes.string,
    urlToImage: PropTypes.string,
    description: PropTypes.string
  }))
};

export default Articles;
