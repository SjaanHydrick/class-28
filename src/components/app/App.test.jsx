import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Article from './Article';
import Articles from './Articles';
import Search from './Search';

describe('App component', () => {

  afterEach(() => cleanup());

  it('renders Articles', () => {
    const articles = [{ title: '', author: '', url: '', urlToImage: '', description: '' }];
    const { asFragment } = render(<Articles articles={articles}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render Article', () => {
    const { asFragment } = render(<Article />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render Search', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});
