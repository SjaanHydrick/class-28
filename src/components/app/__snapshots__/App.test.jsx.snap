// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render Article 1`] = `
<DocumentFragment>
  <figure>
    <a>
      <h2 />
    </a>
    <img />
    <h3 />
    <p />
  </figure>
</DocumentFragment>
`;

exports[`App component render Search 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="searchBar"
      type="text"
    />
    <button
      data-testid="submitButton"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;

exports[`App component renders Articles 1`] = `
<DocumentFragment>
  <div>
    <div
      class="article"
    >
      <figure>
        <a
          href=""
        >
          <h2 />
        </a>
        <img
          src=""
        />
        <h3 />
        <p />
      </figure>
    </div>
  </div>
</DocumentFragment>
`;
